toto/models/matchrule.py

```diff
@@ -0,0 +1,102 @@
+"""
+<Module Name>
+  toto/models/validators.py
+
+<Author>
+  Santiago Torres-Arias <santiago@nyu.edu>
+
+<Started>
+  Nov 18, 2016
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  This module provides functions to ensure the syntax of the matchrules is
+  correct.
+
+"""
+from toto.ssl_commons.exceptions import FormatError
+import toto.ssl_crypto.formats
+
+def _validate_match_rule(keywords):
+  """ private helper to verify the syntax of the MATCH matchrule """
+  MATERIAL_OR_PRODUCT = {'PRODUCT', 'MATERIAL'}
+
+  if not isinstance(keywords, list):
+    raise FormatError("this matching rule is not a list")
```

Comment:
The way you do it is fine. But for the sake of the validator (that checks what should be in the metadata) , I think we should explicitly require lists.